libcxx/utils/libcxx/test/params.py

```diff
@@ -121,6 +136,17 @@ def getStdFlag(cfg, std):
             AddCompileFlag(lambda cfg: getStdFlag(cfg, std)),
         ],
     ),
+    Parameter(
+        name="optimization",
+        choices=["none", "speed", "size"],
+        type=str,
+        help="The version of the standard to compile the test suite with.",
```

Comment:
This help string is not quite accurate. I wonder if it would be simpler to just add a test_flags parameter and pass -O2 from the cache file (similar to #65517)?

Reply:
> This help string is not quite accurate.

Thanks, I will fix that.

> I wonder if it would be simpler to just add a test_flags parameter and pass -O2 from the cache file (similar to https://github.com/llvm/llvm-project/pull/65517)?

That's exactly the thing I don't want to slip towards. In theory it is possible to implement everything else on top of free-form flags being passed to the test suite, but then we lose visibility over what parameters we think make sense to use with the test suite and we'll also start encoding more and more logic into the CMake. The test suite should strive to be independent from the CMake setup, and right now it very much is (with only the small cmake-bridge needed to bridge the two).

Reply:
Thanks for the explanation, that makes sense.